lib/src/compiler/nodes/class.dart

```diff
@@ -2,124 +2,188 @@ part of calzone.compiler;
+        $constructor
+
+        this[$symDartObj][$symBackup] = {};
+        this[$symDartObj][$symJsObj] = this;
+        
+        var proto = mdex.$name.prototype;
+        if(Object.getPrototypeOf(this) !== proto) {
+    """);
+    
+    methodChecks.forEach((check) => output.write(check.toString()));
+    
+    output.write("""
+        }
+      };
+    """);
+
+    final proto = prototype.toString();
+
+    if (proto.length > 0) {
+      // cut off trailing comma
+      output.write("""
+        mdex.$name.prototype = {
+          ${proto.substring(0, proto.length - 1)}
+        };
+      """);
+    }
+  
+    output.write("mdex.$name.prototype[$symIsWrapped] = true;");
+
+    output.write("""
+      mdex.$name.class = function() {
+        function $name() {
+          mdex.$name.apply(this, arguments);
+          console.error('$name.class is deprecated, please change to only use $name');
+        }
+
+        $name.prototype = Object.create(mdex.$name.prototype);
+        $name.prototype["constructor"] = $name;
+
+        return $name;
+    }();
+    """);
+  }
+
   render(Compiler compiler, StringBuffer output) {
-    List<String> names = [];
-    List<StringBuffer> methods = [];
+    // list of function/field names in the class
+    final List<String> names = [];
+    final List<StringBuffer> methods = [];
 
-    StringBuffer constructor = new StringBuffer();
-    StringBuffer prototype = new StringBuffer();
-    StringBuffer global = new StringBuffer();
+    final StringBuffer constructor = new StringBuffer();
+    final StringBuffer prototype = new StringBuffer();
+    final StringBuffer global = new StringBuffer();
 
     _handleClassChildren(Class c, Map memberData, {bool isTopLevel: true}) {
       var mangledFields =
           compiler.mangledNames.getClassFields(c.libraryName, c.name);
       if (mangledFields == null) mangledFields = [];
 
-      List<String> accessors = [];
-      Map<String, Map> getters = {};
-      Map<String, Map> setters = {};
+      final List<String> accessors = [];
+      final Map<String, Map> getters = {};
+      final Map<String, Map> setters = {};
 
-      for (var child in memberData["children"]) {
-        child = child.split("/");
+      final List<Map<String, dynamic>> childData = memberData["children"]
+          .map((f) => compiler.info.getElement(f.split("/")[0], f.split("/")[1]))
+          .toList();
 
-        var type = child[0];
-        var id = child[1];
-
-        var data = compiler.info.getElement(type, id);
-        var name = data["name"];
+      int index = 0;
+      for (final _ in memberData["children"]) {
+        final data = childData[index++];
+        
+        final type = data["kind"];
+        String name = data["name"];
 
         if (type == "function") {
           if (names.contains(name) || name.startsWith("_")) continue;
 
+          // name replacements for operator overloading functions
           if (NAME_REPLACEMENTS.containsKey(data["name"])) {
```

Comment:
NAME_REPLACEMENTS => replacementNames or nameReplacements (Dart Style Guide recommends this)